CorsixTH/Lua/game_ui.lua

```diff
@@ -476,6 +476,15 @@ function GameUI:onCursorWorldPositionChange()
   if not overwindow and wx > 0 and wy > 0 and wx < self.app.map.width and wy < self.app.map.height then
     room = self.app.world:getRoom(wx, wy)
   end
+  -- Find the room associated with the current entity (Usually only applies to doors)
+  if entity and not room then
+    if entity.room then
+      room = entity.room
+    elseif entity.object_type and entity.object_type.class == "SwingDoor" then
```

Comment:
If you check "object_type.id" you get "swing_door_left" (the slave door, in `objects/doors/swing_door_left.lua`) or you get "swing_door_right" (the master door, in `objects/doors/swing_door_right.lua`).

In the latter file is also the SwingDoor class, and it has code to automatically handle differences between slave and master doors, maybe you can use that (not sure).

Reply:
Looks feasible, except for the first 'tick' where only swing_door_right exists to the class.

Reply:
Any further thoughts on this one?

Reply:
Sorry, missed these comments. Yes, using the id sounds much more sensible than using a class name. We only need to check for swing_door_left as a special case because the right-hand door behaves like any other door in a single-door room.